From ea684dae516f1bb6be0f706abf8ed8264a5d420f Mon Sep 17 00:00:00 2001
From: Yann Gautier <yann.gautier@foss.st.com>
Date: Thu, 10 Aug 2023 17:21:07 +0200
Subject: [PATCH 343/841] refactor(stm32mp2): move some functions in
 bl31_plat_arch_setup()

To have a common code between cold and warm boots, move timer and GIC
initialization to bl31_plat_arch_setup(), instead of doing it in
bl31_platform_setup().

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git bdb373418cec507ca5639edfea89857e43044358 refactor(stm32mp2): move some functions in bl31_plat_arch_setup()]
Signed-off-by: Yann Gautier <yann.gautier@foss.st.com>
Change-Id: I9b964196122594cc4dd8dbda7fd7c0cf2ae51de4
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/327283
Tested-by: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
---
 plat/st/stm32mp2/bl31_plat_setup.c | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/plat/st/stm32mp2/bl31_plat_setup.c b/plat/st/stm32mp2/bl31_plat_setup.c
index 5c63b130d..fe30fa9d6 100644
--- a/plat/st/stm32mp2/bl31_plat_setup.c
+++ b/plat/st/stm32mp2/bl31_plat_setup.c
@@ -102,13 +102,14 @@ void bl31_early_platform_setup2(u_register_t arg0, u_register_t arg1,
 
 void bl31_plat_arch_setup(void)
 {
+	generic_delay_timer_init();
+
+	stm32mp_gic_init();
+
 }
 
 void bl31_platform_setup(void)
 {
-	generic_delay_timer_init();
-
-	stm32mp_gic_init();
 }
 
 entry_point_info_t *bl31_plat_get_next_image_ep_info(unsigned int type)
-- 
2.47.2

